test/core/test/expect.test.ts

```diff
@@ -389,3 +389,39 @@ describe('Temporal equality', () => {
     })
   })
 })
+
+describe('expect with custom message', () => {
+  test('built-in matcher throws custom message on failure', () => {
+    expect(() => expect(1, 'custom message').toBe(2)).toThrow('custom message')
+  })
+
+  test('custom matcher with expect.extend throws custom message on failure', () => {
```

Comment:
Maybe let's use this so global expect is not polluted


```suggestion
  test('custom matcher with expect.extend throws custom message on failure', ({ expect }) => {
```